packages/reactive-dict/reactive-dict.js

```diff
@@ -164,6 +164,24 @@ _.extend(ReactiveDict.prototype, {
     
   },
 
+  delete: function(key) {
+    var self    = this;
+    var oldKeys = self.keys;
+
+    _.each(oldKeys, function(value, oldKey) {
+      if (key == oldKey){
+        changed(self.keyDeps[oldKey]);
+        changed(self.keyValueDeps[oldKey][value]);
+        changed(self.keyValueDeps[oldKey]['undefined']);
+        delete self.keys[oldKey]; // clean up
+      }
+    });
+
+    // trigger a change regardless if any matching key found...
+    self.allDeps.changed();
+
```

Comment:
I guess the theory of Tracker is that people should never "expect a rerun" - if you write your autoruns in an idempotent way then there shouldn't be side-effects that you are expecting. It's more about eventual consistency than events.